apps/gitness/src/App.tsx

```diff
@@ -48,17 +39,7 @@ export default function App() {
             <Tooltip.Provider>
               <ExitConfirmProvider>
                 <NavigationProvider routes={routes}>
-                  <RouterContextProvider
-                    Link={Link}
-                    NavLink={NavLink}
-                    Outlet={Outlet}
-                    location={{ ...window.location, state: {}, key: '' }}
-                    navigate={router.navigate}
```

Comment:
This `router.navigate` was seemingly causing unexpected re-routing. It has been replaced with `navigate` function from `useNavigate`.